### PR TITLE
add exception class InvalidGPConstraint

### LINKED ISSUE #828: calling solve w/SignomialEquality
I just accidentally used solve on a model with a SignomialEquality. Instead of getting the usual message about needing to use localsolve I got this message
```
TypeError: SignomialEquality could not simplify to a PosynomialInequality
```
I'm not sure if that is intentional or not, but I found it rather confusing. It took me a while to realize I was using solve not localsolve because I'm accustomed to seeing the usual warning when I make that mistake

### PR DESCRIPTION
**Pull request by bqpd**
closes #828

### CODE REVIEW

**Comment by bqpd** [comment]
@whoburg ready for review

**Comment by bqpd** [comment]
retest this please

File: gpkit/geometric_program.py [hunk 1]
```diff
@@ -66,16 +67,7 @@ def __init__(self, cost, constraints, substitutions=None, verbosity=1):
...
-                constr_posys = constraint.as_posyslt1()
-            except TypeError as err:
-                if err.message == ("SignomialInequality could not simplify to"
-                                   " a PosynomialInequality"):
-                    raise ValueError("GeometricPrograms cannot contain"
-                                     " SignomialInequalities: try forming your"
-                                     " program as SignomialProgram or calling"
-                                     " .localsolve().")
-                raise
+            constr_posys = constraint.as_posyslt1()
```
Comment (whoburg) [inline gpkit/geometric_program.py]: so much cleaner.

File: gpkit/nomials/nomial_math.py [hunk 2]
```diff
@@ -715,8 +718,10 @@ def __init__(self, left, right):
     def as_posyslt1(self):
         "Returns the posys <= 1 representation of this constraint."
         # todo deal with substitutions
-        raise TypeError("SignomialEquality could not simplify to"
-                        " a PosynomialInequality")
+        raise InvalidGPConstraint("SignomialEquality could not simplify"
+                                  " to a PosynomialInequality; try forming"
+                                  " your program as SignomialProgram or"
+                                  " calling .localsolve().")
```
Comment (whoburg) [inline gpkit/nomials/nomial_math.py]: In both the new messages above, what exactly does "try forming your program as SignomialProgram" mean? Should this reference `SignomialsEnabled`?

File: gpkit/tests/t_model.py [hunk 3]
```diff
@@ -259,9 +260,7 @@ def test_trivial_sp(self):
         y = Variable('y')
         with SignomialsEnabled():
             m = Model(x, [x >= 1-y, y <= 0.1])
-        with self.assertRaises(ValueError):
-            # solve should catch the TypeError raised by an SP constraints
-            # and raise its own ValueError instead
+        with self.assertRaises(InvalidGPConstraint):
```
Comment (whoburg) [inline gpkit/tests/t_model.py]: nice

**Comment by whoburg** [comment]
review complete.

File: gpkit/nomials/nomial_math.py (hunk 2 above)
Comment (bqpd) [inline gpkit/nomials/nomial_math.py]: When we're forming an SP (or trying and failing to form a GP from an SP) we don't care about SignomialsEnabled.

**Comment by bqpd** [comment]
Answered re-review. Ready to merge?

**Comment by bqpd** [comment]
Merging.

File: gpkit/nomials/nomial_math.py (hunk 2 above)
Comment (whoburg) [inline gpkit/nomials/nomial_math.py]: yes; what I meant was, what user(s) will encounter this error message, and will it be clear to them what to do next? (e.g. note that `SignomialProgram` is not designed to be user facing)

Comment (bqpd) [inline gpkit/nomials/nomial_math.py]: Oh! Hmm. Any suggestions?

Comment (bqpd) [inline gpkit/nomials/nomial_math.py]: how about `try calling .localsolve instead of solve to form your model as a SignomialProgram`

Comment (whoburg) [inline gpkit/nomials/nomial_math.py]: yes, seems better

Comment (bqpd) [inline gpkit/nomials/nomial_math.py]: Changed on master. :dancer:

**Pull request merged**